pkg/liqoctl/network/cluster.go

```diff
@@ -194,6 +197,123 @@ func (c *Cluster) SetupConfiguration(ctx context.Context, conf *networkingv1beta
+	templateGVR := opts.ServerGatewayType
+
+	s := c.local.Printer.StartSpinner(fmt.Sprintf("Checking gateway client template %s/%s", templateName, templateNamespace))
+
+	template, err := c.checkTemplate(ctx, templateName, templateNamespace, templateGVR)
+	if err != nil {
+		s.Fail(fmt.Sprintf("An error occurred while checking gateway template %s/%s: %v", templateName, templateNamespace, output.PrettyErr(err)))
+		return err
+	}
+
+	if err := c.checkTemplateServerService(template, opts); err != nil {
+		s.Fail(fmt.Sprintf("An error occurred while checking gateway template %s/%s: %v", templateName, templateNamespace, output.PrettyErr(err)))
+		return err
+	}
+
+	s.Success(fmt.Sprintf("Gateway client template %s/%s correctly checked", templateName, templateNamespace))
+	return nil
+}
+
+func (c *Cluster) checkTemplate(ctx context.Context, templateName, templateNamespace, templateGvr string) (*unstructured.Unstructured, error) {
+	// Server Template Reference
+	gvr, err := enutils.ParseGroupVersionResource(templateGvr)
+	if err != nil {
+		return nil, err
+	}
+
+	template, err := c.local.DynClient.Resource(gvr).Namespace(templateNamespace).Get(ctx, templateName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return template, nil
+}
+
+// checkTemplateServerService checks if the GatewayServer service template is correctly set up.
+func (c *Cluster) checkTemplateServerService(template *unstructured.Unstructured, opts *Options) error {
+	switch corev1.ServiceType(opts.ServerServiceType.Value) {
+	case corev1.ServiceTypeClusterIP:
+		return c.checkTemplateServerServiceClusterIP(template, opts)
+	case corev1.ServiceTypeNodePort:
+		return c.checkTemplateServerServiceNodePort(template, opts)
+	case corev1.ServiceTypeLoadBalancer:
+		return c.checkTemplateServerServiceLoadBalancer(template, opts)
+	case corev1.ServiceTypeExternalName:
+		return fmt.Errorf("externalName service type not supported")
+	}
+	return nil
+}
+
+func (c *Cluster) checkTemplateServerServiceClusterIP(_ *unstructured.Unstructured, _ *Options) error {
+	return nil
+}
+func (c *Cluster) checkTemplateServerServiceNodePort(template *unstructured.Unstructured, opts *Options) error {
+	if opts.ServerServiceNodePort == 0 {
+		return nil
+	}
+
+	path := "spec.template.spec.service.spec.ports"
+	templateServicePorts, err := maps.GetNestedField(template.Object, path)
+	if err != nil {
+		return fmt.Errorf("unable to get %s of the server template", path)
+	}
+
+	templateServicePortsSlice, ok := templateServicePorts.([]interface{})
+	if !ok {
+		return fmt.Errorf("unable to cast %s to []interface{}", path)
+	}
+
+	port, ok := templateServicePortsSlice[0].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("unable to cast %s to map[string]interface{}", path)
+	}
+
+	_, err = maps.GetNestedField(port, "nodePort")
+	if err != nil {
+		return fmt.Errorf("unable to get spec.template.spec.service.spec.ports[0].nodePort int the server template, " +
```

Comment:
Once we add a flag to skip this check (as suggested by aleoli), we can add a suggestion of how to skip this check. Morevoer, we might also add a documentation page explaining the concept of template and how to add a field in the template, so that we can add a link to it in the error message